File: 0078-subsets/0078-subsets.py
class Solution:
    def helper(self, l, l1):
        for i in range(len(l)):
            l2 = l[:i] + l[i+1:]

            if l2 not in l1:
                l1.append(l2)
                self.helper(l2, l1)
                
        return l1

    def subsets(self, nums: list[int]) -> list[list[int]]:
        if not nums:
            return [[]]
        else:
            l1 = [nums] 
            
            return self.helper(nums, l1)

File: 0078-subsets/0078-subsets.cpp

class Solution {
public:
    std::vector<std::vector<int>> subsets(std::vector<int>& nums) {
        int n = nums.size();
        int t = 1 << n;
        std::vector<std::vector<int>> p;

        for (int m = 0; m < t; m++) {
            std::vector<int> s;
            for (int i = 0; i < n; i++) {
                if (m & (1 << i)) {
                    s.push_back(nums[i]);
                }
            }
            p.push_back(s);
        }

        return p;
    }
};